psiturk/experiment.py

```diff
@@ -219,6 +219,11 @@ def advertisement():
                (myrule == "pc" and user_agent_obj.is_pc) or\
                (myrule == "bot" and user_agent_obj.is_bot):
                 browser_ok = False
+        elif (myrule == "Safari" or myrule == "safari"):
+            if "Chrome" in user_agent_string:
+                browser_ok = True
```

Comment:
I don't think you want to set `browser_ok` to True, here, because if someone has the rule "chrome,safari", then `browser_ok` will first be set to `False` (for Chrome) and the `True` (because it detects it as Safari, but then Chrome).